atlp-rwanda/script-heroes-bn-backend
# #174797531 enable notification on request

#### What does this PR do?
- Create a notification for the approved request
#### Description of Task to be completed?
- Create models and migration for notifications
- Create controllers for notifications
#### How should this be manually tested?
- Clone this repo `git clone https://github.com/atlp-rwanda/script-heroes-bn-backend.git`
- Switch your working directory `cd script-heroes-bn-backend`
- Checkout this branch `git checkout ft-request-notification-#174797531`
- Create a `.env` file and fill it with `.env-example` data examples
- Install all dependencies `npm install`
- Migrate your database `npx sequelize db:migrate`
- Seed some data in database `npx sequelize db:seed:all
- Start the server `npm run dev`
- Using postman send any request that involves trips
- Still on postman send a get request to `/api/notifications/count`
> You should see the number of all unseen notifications
#### Any background context you want to provide?
- You should log in first to obtain an access token
- Your access token should be added in headers of any request you want to send as `x-auth-token`
- Your profile should be complete to send these requests
- for saving time, you may use users in seed to login because their profiles are already complete
- some endpoints for trip booking
    - `/api/trips/oneway`
    - `/api/trip/return-trip`
    - `/api/trips/multi-city`
- Endpoints related to notifications
    - `/api/notifications`
    - `/api/notifications/id`
- When a request is done, a notification will be created on the side of his/her linemanger
- When a request is approved, a notification will be created on the side of the requester
#### What are the relevant pivotal tracker stories?
[#174797531](https://www.pivotaltracker.com/story/show/174797531)
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/44426710/98081907-99ce1880-1e80-11eb-8ed2-64d44e6a2dea.png)
![image](https://user-images.githubusercontent.com/44426710/98082670-ca628200-1e81-11eb-993c-970b28a361fc.png)
